server/network.go

```diff
@@ -156,19 +154,34 @@ func (lc *localNetwork) createConfig() error {
 			cfg.NodeConfigs[i].ChainConfigFiles[k] = v
 		}
 
-		mergedConfig, err := mergeNodeConfig(local.GetDefaultFlags(), globalConfig, lc.options.customNodeConfigs[nodeName])
-		if err != nil {
-			return fmt.Errorf("failed merging provided configs: %w", err)
+		if cfg.NodeConfigs[i].Flags == nil {
+			cfg.NodeConfigs[i].Flags = map[string]interface{}{}
+		}
+
+		// set flags applied to the specific node
+		var customNodeConfig map[string]interface{}
+		if lc.options.customNodeConfigs[nodeName] != "" {
```

Comment:
Are we sure that lc.options.customNodeConfigs can never be nil here (to avoid a potential panic)?

Reply:
not sure. adding guard